pyro/distributions/sine_bivariate_von_mises.py

```diff
@@ -0,0 +1,280 @@
+            x /= x.norm(dim=0)[None, ...]  # Angular Central Gaussian distribution
+
+            lf = curr_conc[0] * (x[0] - 1) + eigmin[missing > 0] + log_I1(0, torch.sqrt(
+                curr_conc[1] ** 2 + (curr_corr * x[1]) ** 2)).squeeze(0) - phi_den[missing > 0]
+            assert lf.shape == ((missing > 0).sum(), missing[missing > 0].min())
+
+            lg_inv = 1. - curr_b0.view(-1, 1) / 2 + torch.log(
+                curr_b0.view(-1, 1) / 2 + (curr_eig.view(2, -1, 1) * x ** 2).sum(0))
+            assert lg_inv.shape == lf.shape
+
+            accepted = torch.distributions.Uniform(0., torch.ones((), device=conc.device)).sample(lf.shape) < (
+                    lf + lg_inv).exp()
+
+            phi_mask = torch.zeros((*missing.shape, total), dtype=torch.bool, device=conc.device)
+            phi_mask[missing > 0] = torch.logical_and(lengths < (start[missing > 0] + accepted.sum(-1)).view(-1, 1),
+                                                      lengths >= start[missing > 0].view(-1, 1))
+
+            phi[:, phi_mask] = x[:, accepted]
+
+            start[missing > 0] += accepted.sum(-1)
+            missing[missing > 0] -= accepted.sum(-1)
+            max_iter -= 1
+
+        if max_iter == 0 or torch.any(missing > 0):
+            raise ValueError("maximum number of iterations exceeded; "
+                             "try increasing `SineBivariateVonMises.max_sample_iter`")
+
+        phi = torch.atan2(phi[1], phi[0])
+
+        alpha = torch.sqrt(conc[1] ** 2 + (corr * torch.sin(phi)) ** 2)
+        beta = torch.atan(corr / conc[1] * torch.sin(phi))
+
+        psi = VonMises(beta, alpha).sample()
+        phi = phi.view((*self.batch_shape, total))
+        psi = psi.view((*self.batch_shape, total))
+
+        phi_psi = torch.vstack(((phi + self.phi_loc.view((*self.batch_shape, 1)) + pi) % (2 * pi) - pi,
+                                (psi + self.psi_loc.view((*self.batch_shape, 1)) + pi) % (2 * pi) - pi)).T
+        return phi_psi.view(*sample_shape, *self.batch_shape, *self.event_shape)
+
+    @property
+    def mean(self):
+        return torch.stack((self.phi_loc, self.psi_loc), dim=-1)
+
+    @classmethod
+    def infer_shapes(cls, **arg_shapes):
+        batch_shape = torch.Size(broadcast_shape(*arg_shapes.values()))
+        return batch_shape, torch.Size([2])
+
+    def expand(self, batch_shape, _instance=None):
+        try:
+            return super(SineBivariateVonMises, self).expand(batch_shape)
+        except NotImplementedError:
+            validate_args = self.__dict__.get('_validate_args')
+            phi_loc = self.phi_loc.expand(batch_shape)
+            psi_loc = self.psi_loc.expand(batch_shape)
+            phi_conc = self.phi_concentration.expand(batch_shape)
+            psi_conc = self.psi_concentration.expand(batch_shape)
+            corr = self.correlation.expand(batch_shape)
+
+            return type(self)(phi_loc, psi_loc, phi_conc, psi_conc, corr, validate_args=validate_args)
```

Comment:
Could you use the more modern idiom, say [following SoftLaplace]():
```py
def expand(self, batch_shape, _instance=None):
    new = self._get_checked_instance(SineBivariateVonMises, _instance)
    batch_shape = torch.Size(batch_shape)
    new.psi_loc = self.psi_loc.expand(batch_shape)
    new.phi_loc = self.phi_loc.expand(batch_shape)
    new.psi_conc = self.psi_conc.expand(batch_shape)
    new.phi_conc = self.phi_conc.expand(batch_shape)
    new.correlation = self.correlation.expand(batch_shape)
    super(SineBivariateVonMises, new).__init__(batch_shape, self.event_shape, validate_args=False)
    new._validate_args = self._validate_args
    return new
```

Reply:
Yes, of course. I'll update to 
``` python
def expand(self, batch_shape, _instance=None):
  new = self._get_checked_instance(SineBivariateVonMises, _instance)
  batch_shape = torch.Size(batch_shape)
  for k in SineBivariateVonMises.arg_constraints.keys():
    setattr(new, k, getattr(self, k).expand(batch_shape))
  super(SineBivariateVonMises, new).__init__(batch_shape, self.event_shape, validate_args=False)
  new._validate_args = self._valdite_args
  return new
```